merge.cpp

```diff
@@ -61,6 +193,160 @@ void GetSnapshotDim(const int id, const std::string& basename, const std::string
     numSnapshots = S->numColumns();
 }
 
+void GetSnapshotTime(const int id, const std::string& basename, const std::string& varName, std::vector<double> &tSnap)
+{
+    std::string filename = basename + "/param" + std::to_string(id) + "_tSnap" + varName;
+    std::ifstream infile_tSnap(filename);
+    MFEM_VERIFY(infile_tSnap.is_open(), "Snapshot time input file does not exists.");
+
+    tSnap.clear();
+    double t = 0.0;
+    while (infile_tSnap >> t)
+    {
+        tSnap.push_back(t);
+    }
+}
+
+void GetParametricTimeWindows(const int nset, const bool rhsBasis, const std::string& basename, const int windowNumSamples, int &numBasisWindows,
+                              Array<double> &twep, std::vector<std::vector<int>> &offsetAllWindows)
+{
+    std::vector<double> tVec;
+    std::vector<std::vector<double>> tSnapX, tSnapV, tSnapE, tSnapFv, tSnapFe;
+    const int numVar = (rhsBasis) ? 5 : 3;
+    std::vector<int> numSnap(nset*numVar);
```

Comment:
It may be simpler for `numSnap` to be `std::vector<std::vector<int>>`, so that 2D indices can be used.

Reply:
@dylan-copeland After thinking about the whole procedure, I think it is better to keep `numSnap` and `offsetCurrentWindow` to have the same data type and structure, for easier comparison at the final time window, i.e. `numSnap[i] == offsetCurrentWindow[i] + 1`